files/ansible/module_utils/test_result_parser/factory.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+    Base Factory class, with common logic for finding and loading models
+"""
+import os
+import re
+
+try:
+    import test_models.testmodel
+    from test_models.test_model_implementations import *
+except ImportError:
+    import ansible.module_utils.test_result_parser.test_models.testmodel
+    from ansible.module_utils.test_result_parser.test_models.test_model_implementations import *
+
+class TestFactory(object):
+    def __init__(self):
+        self.model_types = {
+                             'bibwtest_openmpi3_model': {'Bi-Directional Bandwidth Test': r'\s+OSU\sMPI\s(\w+[-]\w+\s\w+\s\w+)'},
```

Comment:
This is unnecessary. You can easily insert the "field" in the regex and not have to compare by hand later. ex:

```
{
    'bibwtest_openmpi3_model': r'\s+OSU\sMPI\s-Directional Bandwidth Test'},
    ...
}
````

Reply:
Silly me ahahah